velox/common/base/SimdUtil-inl.h

```diff
@@ -1068,6 +1068,113 @@ struct HalfBatchImpl<
   using Type = Batch64<T>;
 };
 
+template <typename T, typename U, typename A>
+struct ReinterpretBatch {
+#if XSIMD_WITH_SSE2
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::sse2&) {
+    return xsimd::batch<T, A>(data);
+  }
+#endif
+
+#if XSIMD_WITH_AVX
+  static xsimd::batch<T, A> apply(xsimd::batch<U, A> data, const xsimd::avx&) {
+    return xsimd::batch<T, A>(data);
+  }
+#endif
+};
+
+template <typename T, typename A>
+struct ReinterpretBatch<T, T, A> {
+  static xsimd::batch<T, A> apply(xsimd::batch<T, A> data, const A&) {
+    return data;
+  }
+};
+
+#if XSIMD_WITH_NEON
+
+template <typename A>
+struct ReinterpretBatch<uint8_t, int8_t, A> {
+  static xsimd::batch<uint8_t, A> apply(
+      xsimd::batch<int8_t, A> data,
+      const xsimd::neon&) {
+    return vreinterpret_u8_s8(data.data);
+  }
+};
+
+template <typename A>
+struct ReinterpretBatch<uint16_t, int16_t, A> {
+  static xsimd::batch<uint16_t, A> apply(
+      xsimd::batch<int16_t, A> data,
+      const xsimd::neon&) {
+    return vreinterpret_u16_s16(data.data);
+  }
+};
+
+template <typename A>
+struct ReinterpretBatch<uint32_t, int32_t, A> {
+  static xsimd::batch<uint32_t, A> apply(
+      xsimd::batch<int32_t, A> data,
+      const xsimd::neon&) {
+    return vreinterpret_u32_s32(data.data);
+  }
+};
+
+template <typename A>
+struct ReinterpretBatch<uint64_t, int64_t, A> {
+  static xsimd::batch<uint64_t, A> apply(
+      xsimd::batch<int64_t, A> data,
+      const xsimd::neon&) {
+    return vreinterpret_u64_s64(data.data);
+  }
+};
+
+#endif
+
+#if XSIMD_WITH_NEON64
```

Comment:
I think you need to check for NEON64 and if not defined, check NEON. I got below error because NEON64 extends NEON and there is redefinition of the symbol.

```c
SimdUtil-inl.h:1136:8: error: redefinition of ‘struct facebook::velox::simd::detail::ReinterpretBatch<unsigned char, signed char, A>’
 1136 | struct ReinterpretBatch<uint8_t, int8_t, A> {
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
SimdUtil-inl.h:1096:8: note: previous definition of ‘struct facebook::velox::simd::detail::ReinterpretBatch<unsigned char, signed char, A>’
```